test/xmltest/__snapshots__/valid.test.js.snap

```diff
@@ -534,32 +698,28 @@ Array [
 
 exports[`xmltest/valid standalone should match 052.xml with snapshot 1`] = `
 Object {
-  "actual": "<!DOCTYPE doc><doc>𐀀􏿽</doc>",
+  "actual": "<!DOCTYPE doc [
+<!ELEMENT doc (#PCDATA)>
+]>
+<doc>𐀀􏿽</doc>",
   "expected": "<doc>𐀀􏿽</doc>",
 }
 `;
 
 exports[`xmltest/valid standalone should match 053.xml with snapshot 1`] = `
 Object {
-  "actual": "<!DOCTYPE doc><e/>\\"&gt;
-&lt;!ELEMENT doc (e)&gt;
-&lt;!ELEMENT e EMPTY&gt;
-]&gt;
-&lt;doc&gt;&amp;e;",
+  "actual": "<!DOCTYPE doc [
+<!ENTITY e \\"<e/>\\">
+<!ELEMENT doc (e)>
+<!ELEMENT e EMPTY>
+]>
+<doc>&amp;e;</doc>",
   "errors": Array [
-    Array [
-      "error",
-      "element parse error: Error: Hierarchy request error: Only one element can be added and only after doctype",
-      Object {
-        "columnNumber": 2,
-        "lineNumber": 6,
-      },
-    ],
```

Comment:
The current implementation obviously also fixes several other issues: in this case it looks like a totally valid document was previously not parsable but now it is.
Great stuff.